### PR TITLE
Introduce boring type id wrapper

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

**Comment by fppt** [comment]
please retest this please - timeout

**Comment by fppt** [comment]
please retest this please - timeout

File: grakn-core/src/main/java/ai/grakn/concept/TypeId.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public int compareTo(TypeId o) {
+        return getValue().compareTo(o.getValue());
+    }
+
+    @Override
+    public String toString(){
+        return getValue().toString();
+    }
+
+    public boolean isValid(){
```
Comment (aelred) [inline grakn-core/src/main/java/ai/grakn/concept/TypeId.java]: `isEmpty()` might be better, to reflect `TypeId.empty()` (or rename `TypeId.empty()` to `TypeId.invalid()`)

File: grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java [hunk 2]
```diff
@@ -1016,7 +1009,7 @@ private void copyRelation(ResourceImpl main, ResourceImpl<?> other, RelationImpl
         //Gets the other resource index and replaces all occurrences of the other resource id with the main resource id
         //This allows us to find relations far more quickly.
         String newIndex = otherRelation.getIndex().replaceAll(other.getId().getValue(), main.getId().getValue());
-        RelationImpl foundRelation = getTxCache().getModifiedRelations().get(newIndex);
+        RelationImpl foundRelation = getTxCache().getCachedRelation(newIndex);
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java]: What's this change for?

Comment (fppt) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java]: Makes the code shorter. `getCachedRelation` was defined in the cache but this was bypassing it. Functionally it does the same thing.

File: grakn-core/src/main/java/ai/grakn/concept/TypeId.java (hunk 1 above)
Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/concept/TypeId.java]: Agreed. Will change.

**Comment by fppt** [comment]
@aelred review again please

**Comment by fppt** [comment]
please retest this please - timeout

**Comment by fppt** [comment]
please retest this please - timeout

**Comment by fppt** [comment]
please retest this please - timeout

**Comment by fppt** [comment]
please retest this please - timeout

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.graql.analytics.ClusteringTest.testConnectedComponent

Come on!!!!!!

**Comment by fppt** [comment]
please retest this please - timeout

GraqlShellTests dead now

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.controller.GraqlControllerTest.POSTGraqlInsert_InsertWasExecutedOnGraph

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.graql.analytics.ClusteringTest.testConnectedComponentConcurrency 

GOD WHY????

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.controller.GraqlControllerTest.GETGraqlMatchWithGraqlJsonType_ResponseIsCorrectGraql


AHHHHH!!!!

**Comment by fppt** [comment]
please retest this please

**Pull request merged**